octopusdeploy_framework/schemas/git_credential.go

```diff
@@ -42,10 +45,41 @@ func (g GitCredentialSchema) GetResourceSchema() resourceSchema.Schema {
 				Description("The password for the Git credential.").
 				Validators(stringvalidator.LengthAtLeast(1)).
 				Build(),
+			"repository_restrictions": gitCredentialRepositoryRestrictionAttribute(),
 		},
 	}
 }
 
+func gitCredentialRepositoryRestrictionAttribute() resourceSchema.SingleNestedAttribute {
+	return resourceSchema.SingleNestedAttribute{
+		Description: "Sets the repository restrictions associated with the Git credential.",
+		Attributes: map[string]resourceSchema.Attribute{
+			"enabled": util.ResourceBool().
+				Description("Whether repository restrictions are enabled.").
+				Required().
+				Build(),
+			"allowed_repositories": util.ResourceSet(types.StringType).
+				Description("Set of allowed repository URL's.").
+				Required().
+				Build(),
+		},
+		Optional: true,
+		Computed: true,
```

Comment:
I think this is intended from when we spoke through this, but AFAIK this means the existing repository restrictions will be kept when the property is removed entirely from the terraform config. Just wanting to confirm that's intended behaviour. Also might be worth Danys sanity checking me on that...

Reply:
Yes we want the property to still exist - I guess the reason for making it optional is so that existing TF files to keep working. Or should we make this a required property?

Reply:
That makes sense, was just double checking. I'd leave it as optional, a plan should add the default value so customers will be aware of the new property but their existing configs will continue to work.